discovery/generator.go

```diff
@@ -0,0 +1,52 @@
+package discovery
+
+import (
+	"fmt"
+	"regexp"
+)
+
+func incString(s string) string {
```

Comment:
This function also works for increasing hostname, for example `host-[a001:b100]`. I'll add another example to the PR's description.